### PR TITLE
Social: Add polling flag for share status for better UI

### PR DESCRIPTION
**Pull request by manzoorwanijk**
As a part of showing better loading indicator and handle edge cases in post publish panel, we need a flag that indicates that polling for share status is in progress because the current `loading` flag only means that an API request is in flight, it doesn' incorporate the polling interval, i.e. delay in between the API requests.

## Proposed changes:

* Add `polling` flag for share status in social store
* Use polling flag in post publish panel to handle edge cases

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Enable the share status feature - `define( 'JETPACK_SOCIAL_HAS_SHARE_STATUS', true );`
* Create a post and publish it, sharing to some social media connections
* Confirm that the post publish panel reflects the loading correctly - `Sharing to your social media…`
* Create another post and this time share to Instagram without an image or SIG to simulate failure
* Confirm that the failure notice in the post publish panel works correctly
* To simulate timeout, add `sleep( 65 ); // seconds` before `return` in `Automattic\Jetpack\Publicize\REST_Controller::get_post_share_status()`
* Publish a post again and check the post publish panel again.
* Confirm that you see the message, `"The request to share your post is still in progress."`
* To simulate the edge case, add this before `return`
```php
return rest_ensure_response(
	array(
		'shares' => array(),
		'done'   => true,
	)
);
```
* Publish a post again and check the post publish panel again.
* Confirm that you see the message, `"Your post was not shared."`

image
<br>
image

### CODE REVIEW

File: projects/js-packages/publicize-components/src/components/post-publish-share-status/share-status.tsx [hunk 1]
```diff
@@ -52,7 +52,14 @@ export function ShareStatus() {
 		);
 	}
 
+	if ( ! shareStatus.done ) {
+		// If we are here, it means that polling has finished/timedout
+		// but we don't know the share status yet.
+		return <span>{ __( 'The request to share your post is still in progress.', 'jetpack' ) }</span>;
```
Comment (gmjuhasz) [inline projects/js-packages/publicize-components/src/components/post-publish-share-status/share-status.tsx]: Should we leave some guidance here, like "come back later" or "refresh in a few minutes"?

Comment (manzoorwanijk) [reply projects/js-packages/publicize-components/src/components/post-publish-share-status/share-status.tsx]: I was also thinking about the same. Thanks. Adding that to the sentence.

Comment (manzoorwanijk) [reply projects/js-packages/publicize-components/src/components/post-publish-share-status/share-status.tsx]: Done in 5b84870e7c18498fbf8574624cb603ecdf852e39

image

**Review by gmjuhasz — APPROVED** [review]
Looks good!

**Pull request merged**